docs/recipes/bitbucket-pipelines.md

```diff
@@ -0,0 +1,52 @@
+This document shows how you can setup semantic-release with bitbucket's pipelines and a private npm registry.
+
+# Package.json
+Make sure you configure your package.json file with the required plugins.
+A sample configuration may look like this.
+
+```
+"devDependencies": {
+    "@semantic-release/changelog": "^3.0.2",
+    "@semantic-release/commit-analyzer": "^6.1.0",
+    "@semantic-release/git": "^7.0.7",
+    "@semantic-release/npm": "^5.1.3",
+    "@semantic-release/release-notes-generator": "^7.1.4",
+    "semantic-release": "^15.13.2"
+  },
+  "release": {
+    "repositoryUrl": "<SSH Repo URL>",
+    "plugins": [
+      "@semantic-release/commit-analyzer",
+      "@semantic-release/release-notes-generator",
+      "@semantic-release/npm",
+      "@semantic-release/changelog",
+      "@semantic-release/git"
+    ]
+  }
+  ```
+  
+# Pipelines Configuration
+1. (Optional) If you're using a private registry follow the steps here https://npme.npmjs.com/docs/tutorials/pipelines.html then set the environment variables NPM_REGISTRY_URL & NPM_TOKEN in the repository's pipelines' settings from bitbucket's UI..
+2. Add an SSH Key to pipelines settings from the repository's pipelines' settings in bitbucket UI so that it can access the repository.
+3. create a bitbucket-pipelines.yml in the root directory of the repository with the following.
+
+```
+image: node:latest
+
+pipelines:
+  default:
+    - step:
+        caches:
+          - node
+        script: # Modify the commands below to build your repository.
+          # Generates a .npmrc file configured for installing private modules:
+          #
+          # NPM_REGISTRY_URL: the full URL of your private registry
+          #                   defaults to registry.npmjs.org.
+          # NPM_TOKEN: secret token for installing private modules. This
+          #            this token can be found in your .npmrc, after logging in.
+          - printf "//`node -p \"require('url').parse(process.env.NPM_REGISTRY_URL || 'https://registry.npmjs.org').host\"`/:_authToken=${NPM_TOKEN}\nregistry=${NPM_REGISTRY_URL:-https://registry.npmjs.org}\n" >> ~/.npmrc
```

Comment:
The npm plugin already does that. That seems useless here.

Reply:
I tried commenting it out from my pipelines settings but it didnt work, I think since the plugins read from the .npmrc file, this script copies the environment variables (NPM_TOKEN & NPM_REGISTRY_URL) set by the user in the repository's settings to an .npmrc file so it can be read by semantic-release

```
[10:46:32 AM] [semantic-release] › ✖  EINVALIDNPMTOKEN Invalid npm token.
The npm token (https://github.com/semantic-release/npm/blob/master/README.md#npm-registry-authentication) configured in the NPM_TOKEN environment variable must be a valid token (https://docs.npmjs.com/getting-started/working_with_tokens) allowing to publish to the registry https://registry.npmjs.org/.
If you are using Two-Factor Authentication, make configure the auth-only level (https://docs.npmjs.com/getting-started/using-two-factor-authentication#levels-of-authentication) is supported. semantic-release cannot publish with the default auth-and-writes level.
Please make sure to set the NPM_TOKEN environment variable in your CI with the exact value of the npm token.
AggregateError: 
    SemanticReleaseError: Invalid npm token.
        at module.exports (/opt/atlassian/pipelines/agent/build/node_modules/@semantic-release/npm/lib/get-error.js:6:10)
        at module.exports (/opt/atlassian/pipelines/agent/build/node_modules/@semantic-release/npm/lib/verify-auth.js:21:33)
    at /opt/atlassian/pipelines/agent/build/node_modules/semantic-release/lib/plugins/pipeline.js:54:11
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
    at async Object.pluginsConf.<computed> [as verifyConditions] (/opt/atlassian/pipelines/agent/build/node_modules/semantic-release/lib/plugins/index.js:80:11)
    at async run (/opt/atlassian/pipelines/agent/build/node_modules/semantic-release/index.js:82:3)
    at async module.exports (/opt/atlassian/pipelines/agent/build/node_modules/semantic-release/index.js:168:22)
    at async module.exports (/opt/atlassian/pipelines/agent/build/node_modules/semantic-release/cli.js:55:5) {
  name: 'AggregateError'
}
```